repos/iiif-storage/src/Controller/PresleyController.php

```diff
@@ -0,0 +1,168 @@
+                "members" => array_map(function ($manifest) {
+                    $json = $this->builder->build($manifest, false, 0, 1)->getJson();
+                    $json['service'] = [
+                        "highlight" => "info",
+                        "profile" => "row-profile-here",
+                        "values" => [
+                            "",
+                            $json['label'],
+                            "",
+                            "",
+                            "",
+                            ""
+                        ]
+                    ];
+                    return $json;
+                }, $manifests),
+            ]
+            , 200);
+    }
+
+    /**
+     * Gets single derived manifest collection by ID
+     */
+    public function getManifestsCollectionAction()
+    {
+        $this->allowCors();
+
+        $collectionId = $this->params()->fromRoute('collection');
+
+        $manifestItem = $this->repo->getById($collectionId);
+        $manifest = $this->builder->build($manifestItem);
+
+        $manifests = $this->repo->getReplacements($collectionId);
+
+        $json = $manifest->getJson();
+        // Invalid on a collection
+        unset($json['structures']);
+        unset($json['sequences']);
+        $json['@type'] = 'sc:Collection';
+        $json['@id'] = $this->url()->fromRoute(null, [], ['force_canonical' => true], true);
+        $json['members'] = array_map(function($manifest) {
+            return $this->builder->build($manifest, false, 0, 1)->getJson();
+        }, $manifests);
+
+        return new JsonResponse($json);
+    }
+
+    /**
+     * Adds derived manifest to the manifests collection
+     */
+    public function addManifestToCollectionAction()
+    {
+        $request = $this->getRequest();
+        if ($request->isOptions()) {
+            return $this->preflightAction();
+        }
+        $this->allowCors();
+
+        $collectionId = $this->params()->fromRoute('collection');
+        $body = json_decode($this->getRequest()->getContent(), true);
+        $label = $body['label'] ?? 'Untitled manifest';
+        $canvasIds = $body['canvasIds'] ?? [];
+
+        $manifestItem = $this->repo->create(function (ItemRequest $item) use ($label, $collectionId, $canvasIds) {
+            $item->addFields(
+                FieldValue::literalsFromRdf('dcterms:title', 'Label', $label)
+            );
+
+            // Set the manifest URI to internal.
+            $item->addField(
+                FieldValue::url('dcterms:identifier', 'Manifest URI', 'uri:internal')
+            );
+
+            $item->addField(
+                FieldValue::entity('dcterms:replaces', $collectionId, 'resource:item')
+            );
+
+            foreach ($canvasIds as $canvasId) {
+                $item->addField(
+                    FieldValue::entity('sc:hasCanvases', $this->extractCanvasId($canvasId))
+                );
+            }
+        });
```

Comment:
Nice and short implementation. This is not compatible with Presley, instead this is a way of creating a manifest from a label and list of canvas ids and then associating it with a "virtual collection" above.

```
POST /prelsey/collection/{manifest-id}/manifest
{
    "label": "label you want",
    "canvasIds": [
        "https://example.org/iiif/api/canvas/1",
        "https://example.org/iiif/api/canvas/2",
        "https://example.org/iiif/api/canvas/3",
    ]
}
```